sandbox/cgroup/freezer.go

```diff
@@ -187,35 +213,37 @@ func freezeSnapProcessesImplV2(snapName string) error {
 		}
 		return fmt.Errorf("cannot freeze processes of snap %q in group %v", snapName, filepath.Base(dir))
 	}
-	const skipFreezeErrs = false
-	err = applyToSnap(snapName, freezeOne, skipFreezeErrs)
+	// freeze skipping ENOENT errors
+	err = applyToSnap(snapName, freezeOne, os.IsNotExist)
 	if err == nil {
 		return nil
 	}
 	// we either got here because we hit a timeout freezing snap processes
 	// or some other error
-	const skipThawErrs = true
-	thawSnapProcessesV2(snapName, skipThawErrs) // ignore the error, this is best-effort.
+
+	// ignore errors when thawing processes, this is best-effort.
+	alwaysSkipError := func(_ error) bool { return true }
+	thawSnapProcessesV2(snapName, alwaysSkipError)
 	return fmt.Errorf("cannot finish freezing processes of snap %q: %v", snapName, err)
 }
 
-func thawSnapProcessesV2(snapName string, skipErrs bool) error {
+func thawSnapProcessesV2(snapName string, skipError func(error) bool) error {
 	thawOne := func(dir string) error {
 		fname := filepath.Join(dir, "cgroup.freeze")
-		if err := ioutil.WriteFile(fname, []byte("0"), 0644); err != nil && os.IsNotExist(err) {
+		if err := writeFile(fname, []byte("0"), 0644); err != nil && os.IsNotExist(err) {
 			//  the group may be gone already
 			return nil
-		} else if err != nil && !skipErrs {
+		} else if err != nil && (skipError == nil || !skipError(err)) {
```

Comment:
AFAICT we don't test the case where skipError == nil, would be nice to have one.

Reply:
that or just always assume it is not nil

Reply:
> that or just always assume it is not nil

And that's probably the easiest case

Reply:
I've extended the unit test to at least cover applySnap()